### DISCUSSION TITLE
Working addon/integration versions

### DISCUSSION THREAD

**Discussion by tmeuze** [fuatakgun/eufy_security#836] [Q&A] [answered]
Hi, I'm sorry to ask as I understand there is an ongoing issue with 2FA in the latest version. #795 #809 

I've rolled back to bropat's 1.3.5, which is successfully logging in. I am just wondering if a working complimentary version of this integration has been identified - one which does not have the `schema_incompatible` issue (I have tried down to 6.2.0)? Maybe I'm SOL on this, but I really miss this integration. 

Thanks in advance.

**Comment by fuatakgun** [comment, marked as answer]
There are two components; add-on and integration.

Add-on is enabling a communication channel to the integration.

Add-on has its own versioning (1.3.5, 1.4.0 etc) and integration has its own.

When there is something new or breaking on the integration end, it also updates its schema number and to get these new things, integration should tell to add-on about the intended schema number. 

2fa is broken (at least to me) on add-on end after 1.3.5 but never version addon is published with new functionality.

In my latest update on integration, i also bumped the schema version, so if you use the latest version of integration, you must have the latest version of add-on. (I bumped it to get event pictures)

You can always downgrade your integration and/or add-on locally as long as their internal schema numbers are in sync.

**Reply by tmeuze** [reply]
Thank you for your reply!

I realized that I somehow wasn't downgrading the integration properly. I performed a clean reinstall by removing the latest from HACS, rebooting, installing 6.2.0 in HACS, rebooting, then adding the integration. 

Everything (that I need) is working again, with bropat's 1.3.5 addon, and your 6.2.0 integration. 

(I'll be afraid to update for a while now.)

Thank you again!
